pkg/machines/components/vm/vmMigrateDialog.jsx

```diff
@@ -0,0 +1,194 @@
+               isDisabled={copyDisabled}
+               isChecked={storage === "copy"}
+               onChange={() => setStorage("copy")} />
+    );
+
+    if (copyDisabled) {
+        copyRadio = (
+            <Tooltip id="storage-copy-tooltip"
+                     content={_("Offline migration doesn't allow copying storage")}>
+                {copyRadio}
+            </Tooltip>
+        );
+    }
+
+    return (
+        <FormGroup label={_("Storage")}
+                   fieldId="storage"
+                   id="storage"
+                   hasNoPaddingTop>
+            {copyRadio}
+            <Radio id="shared"
+                   name="source"
+                   label={_("I have shared storage set up")}
+                   isChecked={storage === "shared"}
+                   onChange={() => setStorage("shared")} />
+        </FormGroup>
+    );
+};
+
+export const MigrateDialog = ({ vmId, connectionName, toggleModal }) => {
+    const [destUri, setDestUri] = useState("");
+    const [error, setDialogError] = useState({});
+    const [inProgress, setInProgress] = useState(false);
+    const [storage, setStorage] = useState("copy");
+    const [type, setType] = useState("live");
+    const [undefineSource, setUndefineSource] = useState(false);
+    const [validationFailed, setValidationFailed] = useState(false);
+
+    function validateParams() {
+        const validation = {};
+        if (isEmpty(destUri.trim()))
+            validation.destUri = _("Destination URI must not be empty");
+
+        return validation;
+    }
+
+    function onMigrate() {
+        if (!isObjectEmpty(validateParams())) {
+            setValidationFailed(true);
+            return;
+        }
+
+        setInProgress(true);
+        return migrateToUri(connectionName, vmId, destUri, storage, type, undefineSource)
+                .then(toggleModal, exc => {
+                    setInProgress(false);
+                    setDialogError({ dialogError: _("Migration failed"), message: exc.message });
+                });
+    }
+
+    const body = (
+        <Form isHorizontal>
+            <DestUriRow destUri={destUri}
+                        setDestUri={setDestUri}
+                        validationFailed={validationFailed} />
+            <TypeRow type={type}
+                     setType={setType}
+                     setStorage={setStorage}
+                     setUndefineSource={setUndefineSource} />
+            <StorageRow storage={storage}
+                        setStorage={setStorage}
+                        type={type} />
+            <UndefineSourceRow undefineSource={undefineSource}
+                               setUndefineSource={setUndefineSource} />
+        </Form>
+    );
+
+    const footer = (
+        <>
+            {!isObjectEmpty(error) && <ModalError dialogError={error.dialogError} dialogErrorDetail={error.message} />}
+            <Button variant='primary'
+                    isLoading={inProgress}
```

Comment:
We need to either disable the dialog closing while the migration is running or better close it and show a notification that the migration is in progress, and then once more when it's done. Similar approach to the create VM dialog, but we need to communicate it a bit better.